docs/tailwind-css.md

```diff
@@ -0,0 +1,343 @@
+    tw.style(
+      'w-full flex-row items-center justify-between px-4 py-2',
+      pressed && 'bg-pressed',
+    )
+  }
+>
+  Interactive Content
+</Pressable>
+```
+
+❌ **Avoid**
+
+```tsx
+import tw from 'twrnc';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: colors.background.default,
+    padding: 16,
+  },
+});
+
+// Direct twrnc without design system integration
+const styles = tw`bg-default p-4`;
+```
+
+## Style Guidelines
+
+### Platform-Specific Styling Patterns
+
+**React - className usage:**
+
+```tsx
+import { ButtonBase, Icon, IconName } from '@metamask/design-system-react';
+
+// Layout and spacing
+// Interactive states (react supports hover/active)
+<ButtonBase className="h-auto flex-1 flex-col justify-center rounded-lg bg-muted py-4 hover:bg-muted-hover active:bg-muted-pressed">
+  <Icon name={IconName.Bank} className="mb-2" />
+  Buy/Sell
+</ButtonBase>;
+```
+
+**React Native - twClassName and tw usage:**
+
+```tsx
+import { ButtonBase, Icon, IconName, Box, BoxFlexDirection, BoxAlignItems, BoxJustifyContent, Text, FontWeight } from '@metamask/design-system-react-native';
+
+// Custom overrides with twClassName
+<ButtonBase twClassName="h-20 flex-1 rounded-lg bg-muted px-0 py-4">
+  <Box
+    flexDirection={BoxFlexDirection.Column}
+    alignItems={BoxAlignItems.Center}
+    justifyContent={BoxJustifyContent.Center}
+  >
+    <Icon name={IconName.Bank} />
+    <Text fontWeight={FontWeight.Medium}>Buy/Sell</Text>
+  </Box>
+</ButtonBase>
+
+// Interactive states with tw function
+<Pressable
+  style={({ pressed }) =>
+    tw.style(
+      'w-full flex-row items-center justify-between px-4 py-2',
+      pressed && 'bg-pressed',
+    )
+  }
+>
+  Interactive content
+</Pressable>
+
+// Direct tw usage for simple styling
+<Scrollable style={tw`bg-default p-4 rounded-lg`}>
+  Content
+</Scrollable>
+```
+
+## Developer Tools and Configuration
+
+### IDE Integration
+
+- **Enable Tailwind IntelliSense**: Use VSCode with [Tailwind CSS IntelliSense](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) plugin
+- **Configure Workspace**: Follow `.vscode/settings.json` configuration:
+  - Enable string suggestions
+  - Support custom functions: `tw`, `twClassName`, `twMerge`
+
+Example [.vscode/settings.json](https://github.com/MetaMask/metamask-design-system/blob/main/.vscode/settings.json)
+
+### Code Formatting
+
+- **ESLint Integration**: Use the [tailwind eslint plugin](https://github.com/francoismassart/eslint-plugin-tailwindcss)
+- **Consistent Ordering**: Maintain consistent class ordering through prettier-plugin-tailwindcss
+- **Multiple Configs**: Respect the different Tailwind configs for React and React Native
+
+Example [eslint.config.mjs](https://github.com/MetaMask/metamask-design-system/blob/main/eslint.config.mjs)
+
+## Common Pitfalls
+
+### Anti-patterns to Avoid
+
+- **No Arbitrary Values**: Don't use `[]` syntax for arbitrary values unless absolutely necessary
```

Comment:
Maybe not worth doing now but it might be nice to provide examples for all of these guidelines.